tests/jobs/test_local_task_job.py

```diff
@@ -951,11 +953,21 @@ def test_mini_scheduler_works_with_wait_for_upstream(self, caplog, get_test_dag)
         task_k = dag.get_task("K")
         task_l = dag.get_task("L")
         with create_session() as session:
-            ti_k = TaskInstance(task_k, run_id=dr.run_id, state=State.SUCCESS)
-            ti_b = TaskInstance(task_l, run_id=dr.run_id, state=State.SUCCESS)
+            ti_k = dr.get_task_instance(task_k.task_id, session=session)
```

Comment:
The reason for this and other changes in this file is explained in [**Issues identified Point (1)**](https://github.com/apache/airflow/pull/43243#issuecomment-2439774584)